core/api/src/jsonrpc/mod.rs

```diff
@@ -107,6 +110,31 @@ pub trait AxonJsonRpc {
 
     #[method(name = "web3_sha3")]
     async fn sha3(&self, data: Hex) -> RpcResult<Hash>;
+
+    #[method(name = "eth_getTransactionCountByHash")]
+    async fn get_block_transaction_count_by_hash(&self, hash: Hash) -> RpcResult<U256>;
+
+    #[method(name = "eth_getTransactionByBlockHashAndIndex")]
+    async fn get_transaction_by_block_hash_and_index(
+        &self,
+        hash: Hash,
+        position: U256,
+    ) -> RpcResult<Option<Web3Transaction>>;
+
+    #[method(name = "eth_getTransactionByBlockNumberAndIndex")]
+    async fn get_transaction_by_block_number_and_index(
+        &self,
+        number: BlockId,
+        position: U256,
+    ) -> RpcResult<Option<Web3Transaction>>;
+
+    #[method(name = "eth_getStorageAt")]
+    async fn get_storage_at(
+        &self,
+        address: H160,
+        position: Hash,
```

Comment:
```suggestion
        position: U256,
```

The position is not necessarily a `Hash`, it may be a `Uint`. And how to convert the Uint to an `H256`, ref this https://github.com/openethereum/openethereum/blob/main/crates/rpc/src/v1/impls/eth.rs#L1007